### PR TITLE
chore: [SFEQS-1543] Rename whitelist collection

### PR DESCRIPTION
**Pull request by grausof**
Rename issuers whitelist collection name (remove IP) to support also CF whitelist.

### Type of changes

- [ ] Add new resources
- [X] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [X] DEV
- [X] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

Depends on #503 

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/domains/sign/apim.tf [hunk 1]
```diff
@@ -73,12 +73,21 @@ resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_container
   secret              = false
 }
 
-resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_whitelist_collection_name" {
+resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_whitelist_ip_collection_name" {
```
[the remark this answers is not in the document]
Comment (lucacavallaro) [reply src/domains/sign/apim.tf:76]: ok

**Review by lucacavallaro — DISMISSED** [review]

File: src/domains/sign/apim.tf [hunk 2]
```diff
@@ -73,12 +73,21 @@ resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_container
...
+}
+
+resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_whitelist_collection_name" {
+  name                = "io-sign-cosmosdb-issuer-whitelist-collection-name"
+  api_management_name = data.azurerm_api_management.apim_api.name
+  resource_group_name = data.azurerm_api_management.apim_api.resource_group_name
+  display_name        = "io-sign-cosmosdb-issuer-whitelist-collection-name"
+  value               = module.cosmosdb_sql_container_issuer-issuers-whitelist.name
   secret              = false
 }
```
Comment (manuelcoppotelli) [inline src/domains/sign/apim.tf:92]: What is the pourpose of having two named values referencing the same content?

File: src/domains/sign/apim.tf [hunk 3]
```diff
@@ -73,12 +73,21 @@ resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_container
...
 
-resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_whitelist_collection_name" {
+resource "azurerm_api_management_named_value" "io_sign_cosmosdb_issuer_whitelist_ip_collection_name" {
   name                = "io-sign-cosmosdb-issuer-whitelist-ip-collection-name"
```
[the remark this answers is not in the document]
Comment (grausof) [reply src/domains/sign/apim.tf:77]: ditto https://github.com/pagopa/io-infra/pull/504#discussion_r1149354855

**Comment by gunzip** [comment]
Another option is to have two separate collections.

File: src/domains/sign/cosmos_issuer.tf [hunk 4]
```diff
@@ -80,9 +80,9 @@ module "cosmosdb_sql_container_issuer-issuers-by-vat-number" {
   default_ttl = var.io_sign_database_issuer.issuers.ttl
 }
 
-module "cosmosdb_sql_container_issuer-issuers-ip-whitelist" {
+module "cosmosdb_sql_container_issuer-issuers-whitelist" {
   source              = "git::https://github.com/pagopa/terraform-azurerm-v3.git//cosmosdb_sql_container?ref=v4.1.11"
-  name                = "issuers-ip-whitelist"
+  name                = "issuers-whitelist"
```
Comment (manuelcoppotelli) [inline src/domains/sign/cosmos_issuer.tf:85]: ⚠️ Beware that renaming a collection implies destroying the one with the old name and creating a new one with the changed name.

Please make sure to backup the data before applying

**Review by manuelcoppotelli — APPROVED** [review]
I expect another change to follow in order to remove the named value no more necessary.

**Pull request merged**